Author: Apollon Oikonomopoulos <apoikos@debian.org>
Description: Allow compilation with mtl 2.1
 Provide runExceptT using the primitives in tansformers 0.3. This allows us to
 lower the dependency of mtl to 2.1 and compile with GHC 7.6.
Last-Update: 2014-10-11
Forwarded: no
--- a/xmobar.cabal
+++ b/xmobar.cabal
@@ -110,8 +110,9 @@
       unix,
       time,
       filepath,
+      transformers,
       X11 >= 1.6.1,
-      mtl >= 2.2.1 && < 2.3,
+      mtl >= 2.1 && < 2.3,
       parsec == 3.1.*,
       HTTP >= 4000,
       stm >= 2.3 && < 2.5
--- a/src/Bitmap.hs
+++ b/src/Bitmap.hs
@@ -18,7 +18,6 @@
 
 import Control.Applicative((<|>))
 import Control.Monad
-import Control.Monad.Except(MonadError(..), runExceptT)
 import Control.Monad.Trans(MonadIO(..))
 import Data.Map hiding (foldr, map, filter)
 import Graphics.X11.Xlib
@@ -33,6 +32,18 @@
 import XPMFile(readXPMFile)
 #endif
 
+#if MIN_VERSION_mtl(2, 2, 1)
+import Control.Monad.Except(MonadError(..), runExceptT)
+
+#else
+import Control.Monad.Error(MonadError(..))
+import Control.Monad.Trans.Error(ErrorT, runErrorT)
+
+runExceptT :: ErrorT e m a -> m (Either e a)
+runExceptT = runErrorT
+
+#endif
+
 data BitmapType = Mono Pixel | Poly
 
 data Bitmap = Bitmap { width  :: Dimension
--- a/src/XPMFile.hsc
+++ b/src/XPMFile.hsc
@@ -13,7 +13,11 @@
 
 module XPMFile(readXPMFile) where
 
+#if MIN_VERSION_mtl(2, 2, 1)
 import Control.Monad.Except(MonadError(..))
+#else
+import Control.Monad.Error(MonadError(..))
+#endif
 import Control.Monad.Trans(MonadIO(..))
 import Graphics.X11.Xlib(Dimension, Display(..), Drawable, Pixmap)
 import Foreign.C.String(CString, withCString)
